hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/LegacyReplicationManager.java

```diff
@@ -1613,6 +1614,18 @@ private boolean isOpenContainerHealthy(
         .allMatch(r -> compareState(state, r.getState()));
   }
 
+  private void setHealthStateForClosing(Set<ContainerReplica> replicas,
+                                        ContainerInfo container,
+                                        ReplicationManagerReport report) {
+    if (replicas.size() == 0) {
+      report.incrementAndSample(HealthState.MISSING, container.containerID());
+      report.incrementAndSample(HealthState.UNDER_REPLICATED,
+              container.containerID());
+      report.incrementAndSample(HealthState.MIS_REPLICATED,
+              container.containerID());
```

Comment:
Why is the container categorized in all of these states?  Isn't `MISSING` enough?

Reply:
In theory, MISSING could be enough but we wanted to have the same behavior as we have in Recon - this is way we have set all the states.

Reply:
> Isn't MISSING enough?

This is also consistent with the replication manager detecting and reporting "`MISSING`" when the container is in the `CLOSED` state.